### ISSUE TITLE
omegaconf.errors.ConfigAttributeError: Missing key pretty

### ISSUE THREAD

**Issue by xieliang11111** [qq456cvb/Point-Transformers#21]
请问这个问题该怎么解决
![image](https://user-images.githubusercontent.com/22213368/140026532-50867ac1-10a3-4099-afdc-ac45869ebb4f.png)

**Comment by lvhuanhuan123** [comment]
请问这个问题你解决了嘛？谢谢

**Comment by hengyu2333** [comment]
I have deal with this problem. The torch version should fit with your cuda version. My cuda is 10.0, so i need to choose the torch version==1.12.0

# CUDA 10.0 this is find in pytorch web
conda install pytorch==1.2.0 torchvision==0.4.0 cudatoolkit=10.0 -c pytorch

**Comment by hengyu2333** [comment]
It needs to use hydra==1.0
#pip install hydra==1.0 
#need to detect the GPU device

**Comment by struggle2893** [comment]
Has your problem been resolved? If it is solved, can you explain where the problem is and how to solve it?

**Comment by struggle2893** [comment]
> 



> 它需要使用hydra==1.0#pip安装hydra==1.0#need来检测GPU设备

I tried "pip install hydra-core==1.0.1" and the installation was successful, but there are still problems. as follows:
D:\Anaconda\lib\site-packages\hydra\core\utils.py:143: UserWarning: register_resolver() is deprecated.
See https://github.com/omry/omegaconf/issues/426 for migration instructions.

   OmegaConf.register_resolver(name, f)
Traceback (most recent call last):
   File "train_cls.py", line 52, in main
     print(args.pretty())
omegaconf.errors.ConfigAttributeError: Missing key pretty
     full_key: pretty
     object_type=dict

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

Can you help me solve the problem? Thanks!

**Comment by hengyu2333** [comment]
yes, i have dealed with the problem. you need to annotate or delete this line " print(args.pretty())" . I cannot find what the problem with it, i cannot find out the key-word of pretty. It maybe also problem with another thing....

**Comment by struggle2893** [comment]
> yes, i have dealed with the problem. you need to annotate or delete this line " print(args.pretty())" . I cannot find what the problem with it, i cannot find out the key-word of pretty. It maybe also problem with another thing....

Thank you very much for your answer, this problem has been solved!

**Comment by qq456cvb** [comment]
Fixed on `master`. Root cause: `OmegaConf.pretty()` was removed in omegaconf 2.x (pulled in by current Hydra), and because the script disables struct mode, `args.pretty` was being read as a missing config key. I replaced `print(args.pretty())` with `print(OmegaConf.to_yaml(args))`, so a fresh clone now works without any edits. Thanks @hengyu2333 for the workaround. Closing — feel free to reopen if it persists.
